### ISSUE TITLE
[Units] New Units from UMC/Mainline Campaigns to Core Discussions

### ISSUE THREAD

**Issue by knyghtmare** [wesnoth/wesnoth#7335]
Opening this here since I have noticed several new units being added to core from mainline and UMC campaigns (like there's no tomorrow).
Why? To discuss what other possible candidates can be.

Starting off:
1. Monster Crab (UtBS)
2. Cave Wyrmlet and Cave Wyrm (SoF)
3. Eyestalk (TSG)
4. Tauroch (UtBS)
5. Dustbok (UtBS)
6. Young Gryphon (TSoG)
7. Elder Gryphon (AWoF)

Might be more, but we leave that to the rest of the community to present here.

**Comment by Elvish-Hunter** [comment]
I remember that someone suggested renaming the Young Gryphon to Gryphlet, so there's this aspect to consider.
We might also add back the Cockatrice: it was in core until the 1.2 series, then it was removed along with two scenarios from HttT. It already had new sprites (made by Jetrel, IIRC), so it only lacks a portrait. It might also need a rework of the petrification ability.

**Comment by knyghtmare** [comment]
> I remember that someone suggested renaming the Young Gryphon to Gryphlet, so there's this aspect to consider.
We might also add back the Cockatrice: it was in core until the 1.2 series, then it was removed along with two scenarios from HttT. It already had new sprites (made by Jetrel, IIRC), so it only lacks a portrait. It might also need a rework of the petrification ability.

Added as number 8 and also edited number 6 with the name/ID change option

**Comment by CelticMinstrel** [comment]
Yes, I think having a core unit that actually uses petrification would be good.

I have no real objections to any of the named ones. Eyestalk is the only slightly questionable one of the bunch, but whatever.

I will note that, generally speaking, I think units representing specific characters should not be moved to core. There are none listed in the opening post though.

**Comment by Hejnewar** [comment]
> along with the entire Quenoth
I will quickly mention that Im currently preparting to get Quenoth rework up soon (might take a while depending on my work shedule) 

Also I will propose Bone Bird (SotA) with UR stats.

**Comment by CelticMinstrel** [comment]
Bone bird is a L1 chocobone, I guess? I agree that would be good to move to core.

**Comment by knyghtmare** [comment]
I was looking for the Bone Bird but could not find it...and then I see this 
```ini
hide_help=yes
    do_not_list=yes
```
> Also I will propose Bone Bird (SotA) with UR stats.

Okay, seems like a good fit

**Comment by CelticMinstrel** [comment]
How does the Ultimate Recruits bone bird differ from the SotA one? Just numbers?

**Comment by CelticMinstrel** [comment]
I guess WoF has a dolphin and the stymphalian. Not sure about the sprites though.

The Wyrm is also used in WoF so that might be a good reason to move it to core.

**Comment by knyghtmare** [comment]
https://units.wesnoth.org/1.16/Ultimate_Recruits/en_US/Bone%20Bird%20UR.html

The SotA makes it only a "dummy unit" with no attacks and the UR one adds stats

**Comment by knyghtmare** [comment]
> I guess WoF has a dolphin and the stymphalian. Not sure about the sprites though.
The Wyrm is also used in WoF so that might be a good reason to move it to core.

SoF also has a red "ranged" variant which is a IPF version of the green one. That is also something to consider.

**Comment by Elvish-Hunter** [comment]
How about worms? I have a Giant Worm and Cerberus Worm sprite (made by battlestar) in TSoG; once again, they lack a portrait and might need balancing.

**Comment by knyghtmare** [comment]
> How about worms? I have a Giant Worm and Cerberus Worm sprite (made by battlestar) in TSoG; once again, they lack a portrait and might need balancing.

_remembers that scenario of worm zerg swarm attack in RftA_

**Comment by Hejnewar** [comment]
> How does the Ultimate Recruits bone bird differ from the SotA one? Just numbers?
Sorry for being unclear. 

I think gave it an attack becaue I think it doesnt have any by deafult, other than that just numbers.

**Comment by CelticMinstrel** [comment]
> The SotA makes it only a "dummy unit" with no attacks and the UR one adds stats

Okay, that looks good, but it should also advance to Chocobone in my opinion.

> How about worms? I have a Giant Worm and Cerberus Worm sprite (made by battlestar) in TSoG; once again, they lack a portrait and might need balancing.

Do we have a desert worm with sand submerge? There was one in Oath of Allegiance, though the sprite is pretty terrible. (It also didn't have sand submerge.)

**Comment by knyghtmare** [comment]
> Okay, that looks good, but it should also advance to Chocobone in my opinion.

I am not sure why this should be done. The level 1 is 4-3 melee blade charge and the chocobone has a rider and a 9-2 melee pierce charge. Seems very odd.

> Do we have a desert worm with sand submerge? There was one in Oath of Allegiance, though the sprite is pretty terrible. (It also didn't have sand submerge.)

Wont that giant worm be too big for the sand submerge? Kind of the same situation as the skeletal dragon and submerge.

**Comment by Hejnewar** [comment]
I also dont think that this is great or even needed advancement given the distinct scenarios these two would appear in.

**Comment by CelticMinstrel** [comment]
> Wont that giant worm be too big for the sand submerge? Kind of the same situation as the skeletal dragon and submerge.

No. It's a worm. It submerges not by conventional digging but by swallowing the sand and expelling it out the other end.

But the specific worm in Oath of Allegiance may not be as good for that. I thought I saw a better one somewhere but I can't remember where.

**Comment by Hejnewar** [comment]
> What? But they're clearly variations of the same creature and are level 1 and level 2 respectively. Why wouldn't you make it advance?

Honestly for the same reason why I wouldnt make horse advance to horserider. I do understand that this is an undead and it doesnt work this way most likely but I still see Chokobone as a army unit while I see Bone Bird as much more of an animalistic unit that can even naturally appear in forests of death or something like that.

**Comment by CelticMinstrel** [comment]
> I see Bone Bird as much more of an animalistic unit that can even naturally appear in forests of death or something like that.

"animalistic"

"forests of death"

Uhh. Okay then?

That said, you do have a point about the horse to horse rider thing. I forgot the chocobone had a rider. I'll retract my request for an advancement.

**Comment by knyghtmare** [comment]
Seems we have 17 items in the list already.

<img width="373" alt="Screenshot 2023-01-29 at 3 56 55 PM" src="https://user-images.githubusercontent.com/5283677/215313206-dfc426ae-ef71-4ba9-a5d3-127054b2e689.png">

**Comment by demario-w** [comment]
Consider putting new units that are not visible in mainline scenarios in their own text domain for translation. Adding them up to the #wesnoth-units domain will have the statistics for unmaintained translations to drop for that domain.

The #wesnoth-units domain is one of core domains to wesnoth.
When failing to reach 80% coverage (or dropping below that level from addition of new strings) on all core domains, languages are dropped from the list of languages available for selection in the UI. This is bad for native speakers (see "how to enable my language" in the forum).
This is also bad for potential new translators: they don't know which strings to focus on to improve the translation in mainline scenario -- translation of UMC units won't be visible.

**Comment by knyghtmare** [comment]
> Consider putting new units that are not visible in mainline scenarios in their own text domain for translation. Adding them up to the #wesnoth-units domain will have the statistics for unmaintained translations to drop for that domain.

So, to my understanding, are you suggesting these new units to be placed in some trial/intermittent textdomain to avoid manipulation of `#textdomain wesnoth-units`?

I suppose this issue tracker can discuss it.

**Comment by stevecotton** [comment]
> The #wesnoth-units domain is one of core domains to wesnoth. When failing to reach 80% coverage (or dropping below that level from addition of new strings) on all core domains, languages are dropped from the list of languages available for selection in the UI.

While adding them to `#wesnoth-units` does mean more strings are needed to reach the 80% mark, I think it's still better to put them in that core textdomain. Each unit is likely to add 2-4 short strings for name and attack names, and 1 long one for the description - the only two languages that close to the 80% line are Galician and Ukrainian, neither of which has seen updates since 2019, and they're both already below 80% on master.

Also, these units would be available for any mainline scenario to use; so it's likely that they'll be added to more scenarios before 1.18 releases.

**Comment by Pentarctagon** [comment]
Keep in mind that any units that get their stats changed when moving from a mainline campaign to core need to be approved by nemaara.

**Comment by knyghtmare** [comment]
> It might also need a rework of the petrification ability.

Now that I did some research. I dont think we can consider Cockatrice as reworking Petrification at the engine level will break _Invasion from the Unknown_ (one of my favorite UMCs). There is one scenario where one mini-boss and his spawns use this extensively and the scenario is built around it so, I rather not commit an atrocity (breaking IftU) of such a level.

**Comment by Elvish-Hunter** [comment]
> I dont think we can consider Cockatrice as reworking Petrification at the engine level will break Invasion from the Unknown

We can still consider it: we just need to leave petrification as it is now. The issue was that such ability was (is?) considered fundamentally broken (there's no default way to remove the petrified status, unlike slow and poison; however, the same can be said for the unhealable status). As always, it'll be up to scenario authors to handle petrification in adequate ways.

**Comment by doofus-01** [comment]
This is way too much being added, unless these have portraits and basic animations (at minimum).

I think I'd considered the centipedes for 1.16, I recall they have good animations.  I don't remember what the issue was that kept them out, I'll have to review old messages and notes.

If there is some bullpen that these new units can be moved to that isn't default core (I guess it's similar to the textdomain request), I wouldn't have any objections.  Same for the other units being shuffled in various PRs.

**Comment by CelticMinstrel** [comment]
> If there is some bullpen that these new units can be moved to that isn't default core (I guess it's similar to the textdomain request), I wouldn't have any objections. Same for the other units being shuffled in various PRs.

Hmm… I suppose we could lock them behind a mainline `[resource]`, call it something like "experimental_units"?

**Comment by stevecotton** [comment]
> "experimental_units"

If we added a version number in it, "experimental_units_1_17_13" or "experimental_units_2023q1" (first quarter of the year) then:
* any campaign that used them could easily check what the differences were to the final version
* it would be easy to make a "backport addon" for 1.16 that other add-ons could use the resources from

I'm wondering about that idea of a "backport addon" for 1.16 for the units already moved to core too. So add-ons for 1.16 could depend on it, load the units from it for 1.16, and that part of porting to 1.18 can already be done in the 1.16 cycle.

**Comment by Pentarctagon** [comment]
What's the dividing line for what would get moved to core vs a mainline [resource] vs left in a campaign?

**Comment by knyghtmare** [comment]
> What's the dividing line for what would get moved to core vs a mainline [resource] vs left in a campaign?

Core: if decent animations + portrait

mainline (resource-locked): animations (if any) + highly used in both mainline campaigns and UMC

left as is: special unit (one time scenario appearance), requires special mechanics specific to scenario

perhaps this?

I am more inclined in allowing a Wesnoth-Devs addon "Curated Unit Pack" be allowed to on the addon server and UMC can add dependency to that.

**Comment by knyghtmare** [comment]
Also, I think the unit addition discussions should have a thread on forums to allow community suggestions and opinions.
To remove the ideation that "Devs dont care about community"

**Comment by CelticMinstrel** [comment]
> What's the dividing line for what would get moved to core vs a mainline [resource] vs left in a campaign?

I'd say leave them in a campaign if they're a unit representing a single character (or a very limited number of characters). Quenoth Youth is one example of this; I believe the Dwarvish Annalist line is as well. Also if it's a one-off monster it can stay in the campaign, unless it's popular or something.

I think any unit that's used in more than one mainline campaign should be moved to core (potentially behind a `[resource]`). Similarly any unit that is fairly "generic" such as the Naga Hunter should be moved to core (again potentially behind a `[resource]`).

As for moving from a mainline `[resource]` into core proper, I guess that'd be up to doofus-01.

**Comment by cooljeanius** [comment]
I'd just like to add the quintain from the tutorial here

**Comment by knyghtmare** [comment]
> I'd just like to add the quintain from the tutorial here

How'd we miss this?

**Comment by stevecotton** [comment]
The quintain's combat stats have it animated by magic with magical attacks. It feels like a weird object, that would need a wizard or necromancer near it to make it move. It's only got a single 2kB .png image, so having a copy in core wouldn't save much of a an add-on campaign's size.

I feel it doesn't belong in core - I'd expect anyone using it in a campaign to copy the tiny image and edit the .cfg to fit with their scenario.

**Comment by CelticMinstrel** [comment]
By the way, I'd also be quite interested to see a living version of the Bone Bird. Obviously it can't be a chocobo for copyright reasons, but there are plenty of real giant flightless birds to draw on instead, such as the extinct American Terror Bird, or the ostrich, or the emu.

**Comment by knyghtmare** [comment]
>  I'd also be quite interested to see a living version of the Bone Bird

![terror-owl](https://user-images.githubusercontent.com/5283677/221490952-ff3407dd-b5c5-44e4-b0ec-ecb0e2102d9a.png)

**Comment by Elvish-Hunter** [comment]
I see that the Rabbit hasn't been proposed yet; it was one of the things that doofus-01 planned to work on: https://forums.wesnoth.org/viewtopic.php?p=663690#p663690. In the add-ons servers there are an old version and a new version, which was made by Shield: https://forums.wesnoth.org/viewtopic.php?p=588729#p588729.

**Comment by knyghtmare** [comment]
> I see that the Rabbit hasn't been proposed yet; it was one of the things that doofus-01 planned to work on

your token number is 23

**Comment by cooljeanius** [comment]
> I see that the Rabbit hasn't been proposed yet; it was one of the things that doofus-01 planned to work on: [forums.wesnoth.org/viewtopic.php?p=663690#p663690](https://forums.wesnoth.org/viewtopic.php?p=663690#p663690). In the add-ons servers there are an old version and a new version, which was made by Shield: [forums.wesnoth.org/viewtopic.php?p=588729#p588729](https://forums.wesnoth.org/viewtopic.php?p=588729#p588729).

Oh yeah, rabbits are something I've been wishing to have occasionally

**Comment by CelticMinstrel** [comment]
Oh yes, Rabbits are needed for one of the AI demo scenarios!

**Comment by Jonathan-Kelly** [comment]
> I think I'd considered the centipedes for 1.16, I recall they have good animations. I don't remember what the issue was that kept them out, I'll have to review old messages and notes.

@doofus-01
How would you rate the animation and base sprite qualities of the UtBS Crab and Demon monsters? And the UtBS Orb "alien"? Wondering if these are just lacking portraits or if they have sprite problems too.

**Comment by Gothyoba** [comment]
I’d add owls (https://units.wesnoth.org/1.18/Assault_on_the_Citadel/en_US/Owl.html, https://units.wesnoth.org/1.18/Assault_on_the_Citadel/en_US/Great%20Owl.html, previously proposed on forums in https://forums.wesnoth.org/viewtopic.php?p=664374#p664374) and the Dwarvish Witness line (THoT, lacks protrait, though Angarthing’s protrait can be used) to this.

**Comment by CelticMinstrel** [comment]
I don't think I'd add the Dwarvish Witness line. From what I recall, it was basically a hero unit? A hero unit line doesn't really qualify to be added, in my opinion.

**Comment by knyghtmare** [comment]
List updated for Owl and Great Owl. Not taking Dwarvish Witness into consideration. A hero unit should stay a hero unit. Unless it's used by multiple mainline campaigns (as is case of Elder Mage and Orcish Leader)

**Comment by Gothyoba** [comment]
While as far as I’m aware they are not used elsewhere in mainline, they seem to be in use in some UMC according to the Wesnoth Units Database:

https://www.wesnoth.org/units/1.18/Struggle_For_Freedom/en_US/Dwarvish%20Witness.html

https://www.wesnoth.org/units/1.18/Through_the_Dwarven_Roots/en_US/Dwarvish%20Witness.html

Also, their unit descriptions seem to imply that there are many such dwarves, and that they fulfill a very important role in Dwarvish society that is not filled in by other units:

A Witness functions as the eyes of the dwarves’ history, a deep lore that they never share with outsiders. The presence of a Witness inspires dwarvish warriors with the knowledge that their deeds (and their deaths) will not go unrecorded. They learn a fighting style deliberately unlike that of their fellows, one designed to turn the vaunted strength of the dwarves against itself. The person of a Witness is considered sacred, and Witnesses are often used as envoys between dwarvish clans.

Dwarvish Annalists are the historians of this ancient and proud race. They are few in number, and their very existence is not generally known to the other peoples of the Great Continent. To the abilities of the Witness they add, through the study of herb and mineral lore, the ability to cure poison.

Loremasters are the sages of the dwarvish race. Their counsel is greatly valued by war-leaders and chieftains, and they are viewed with awe by ordinary dwarves, for they alone have plumbed the deepest and most closely-held secrets of dwarven knowledge. It is said that where a loremaster stands, there is the soul of the dwarves. To the abilities of the Annalist they add a limited ability to heal.

Not a single unit other than the Dwarvish Loremaster heals among the Dwarvs, which must make them veyr essential to the dwarves. Personally, this makes them feel less like a specific hero unit and more like an important yet small group like other unit (e.g. Runesmiths). IIRC they also have good animations. I don’t feel very strongly about these, but I do think they should be considered further.

**Comment by Gothyoba** [comment]
I compiled some proposals made in the Wesnoth Fauna thread (https://forums.wesnoth.org/viewtopic.php?t=52894) not added here:

https://www.wesnoth.org/units/1.18/A_Song_of_Fire/en_US/Ice%20Ghost.html - Ice Ghost
https://units.wesnoth.org/1.16/Snowdrop_of_Dawn/en_US/Cave%20Worms.html - Cave Worms
https://www.wesnoth.org/units/1.18/War_of_Legends/en_US/Sea%20Wyrm.html - Seawyrm
https://units.wesnoth.org/1.14/Oath_of_Allegiance/en_US/Magical%20Scimitar.html - Magical Scimitar
https://www.wesnoth.org/units/1.14/The_Ravagers/en_US/Chicken%20TRS.html - Chicken
https://www.wesnoth.org/units/1.14/The_Ravagers/en_US/Pig%20TRS.html - Pig
https://www.wesnoth.org/units/1.10/AI-demos/en_US/Deer.html - Deer
https://www.wesnoth.org/units/1.14/The_Ravagers/en_US/Stag%20TRS.html - Stag
https://units.wesnoth.org/1.14/Ageless_Era/en_US/AE_arc_khthon_Ram.html - Ram
https://units.wesnoth.org/1.14/Ageless_Era/en_US/AE_arc_ukians_Ukian_Seal.html - Seal
https://units.wesnoth.org/1.12/Extended_Armies_Era/en_US/EAE%20Shessinsaan.html - Shessinsaan
https://www.wesnoth.org/units/1.10/Invasion_from_the_Unknown/en_US/Giant%20Leech.html - Giant Leech
https://www.wesnoth.org/units/1.10/Invasion_from_the_Unknown/en_US/Brain%20Drainer.html - Brain Drainer
https://units.wesnoth.org/1.18/WarOfTheGods/en_US/WOTG_Leviathan.html - Leviathan
https://units.wesnoth.org/1.18/WarOfTheGods/en_US/WOTG_Giant%20Sea%20Turtle.html - Giant Sea Turtle
https://units.wesnoth.org/1.18/WarOfTheGods/en_US/WOTG_Sea%20Turtle.html - Sea Turtle
https://units.wesnoth.org/1.18/WarOfTheGods/en_US/WOTG_Baby%20Sea%20Turtle.html - Baby Sea Turtle
https://units.wesnoth.org/1.16/A_Song_of_Fire/en_US/Black%20Cat.html - Black Cat, could be added to the new cats race
https://units.wesnoth.org/1.16/A_Song_of_Fire/en_US/Jinx%20Beast.html - Jinx Beast
https://units.wesnoth.org/1.16/A_Song_of_Fire/en_US/Calamity.html - Calamity

**Comment by CelticMinstrel** [comment]
For the leviathan to be a turtle is _very_ weird. Normally a leviathan would be either a whale or a dragon, or maybe a serpent. It's not _completely_ out there, as a turtle is at least a "sea monster", but…

**Comment by knyghtmare** [comment]
> Jinx beast and calamity seem too specialized to be worth mainlining. I guess that covers the black cat, too. It's not really a cat at all, actually – it's a chimera.

I agree too. They are part of an MP faction and should stay as such. You would only break several UMC in the process.

> Ice ghost seems like a fine counterpart to the existing fire elemental

Would be nice if it had a nice standing animation.

> Shessinsaan

What even is this?

**Comment by Gothyoba** [comment]
I agree with most of your points here. The majority of these don’t seem like oarticularly good proposals. The Black Cat/Jinx Cat/Calamity line was proposed by myself, though I agree mainlining them is probably not a good idea. The Brain Drainer personally looks interesting. It seems something similar to that was seriously prposed by doofus-01 fro 1.18 and could get into 1.20 in my opinion. As for the Leviathan, I think it should be renamed, though it would be good to consider it, together with the other turtles. Pigs/chickens probably need new art, but would probably be a pretty good addition if their tweaked a bit.

**Comment by CelticMinstrel** [comment]
> Pigs

What? We already have pigs.

**Comment by Gothyoba** [comment]
> What? We already have pigs.

We have boars, but I don’t think we have pigs. However, their not that different, so I suppose adding mroe similar creatures would likely only add unnecessary clutter.

Shesinssaans seem to be from the Extended Armies Era’s Monster Horde Faction, and have not been used much elsewhere.

While I don’t personally agree on adding these, I have found this unit line suggested from the froum thread above:

https://www.wesnoth.org/units/1.18/Extended_Armies_Era/en_US/EAE%20Pyrophorus.html
https://www.wesnoth.org/units/1.18/Extended_Armies_Era/en_US/EAE%20Great%20Worm.html
https://www.wesnoth.org/units/1.18/Extended_Armies_Era/en_US/EAE%20Giant%20Worm.html
https://www.wesnoth.org/units/1.18/Extended_Armies_Era/en_US/EAE%20Cerberus%20Worm.html

**Comment by CelticMinstrel** [comment]
> We have boars, but I don’t think we have pigs. However, their not that different, so I suppose adding mroe similar creatures would likely only add unnecessary clutter.

…what. Boar literally means "male pig". They aren't just similar – they are exactly the same thing.

**Comment by Gothyoba** [comment]
I think a boar can be a male pig or the wild ancestor of domesticated pigs.

**Comment by knyghtmare** [comment]
> https://www.wesnoth.org/units/1.14/The_Ravagers/en_US/Pig%20TRS.html - Pig
>…what. Boar literally means "male pig". They aren't just similar – they are exactly the same thing.

Let me clarify one thing. The Pig and Gorer were added initially during the 1.15.x cycle. 
However the IDs were changed to be Woodland Boar and Piglet after it was complained by a UMC author that the unit line broke in his addon because of the unit IDs. So, they have been added already. Check the Git history of the 2 units.

**Comment by doofus-01** [comment]
There's nothing wrong with discussions, but be aware that finding some sprite that could be a nice unit to have in core isn't really helpful unless there is a concomitant effort to work on whatever changes are needed and to flesh out the animations.  There could always be exceptions if there is some clear gap being filled or the sprite/animations are really impressive.

I realize it's hard to commit unpaid time and effort to something that might not go anywhere, but new core units generally need artists and effort, not internet archeologists and scavenging.

**Comment by Jonathan-Kelly** [comment]
> the unit line broke in his addon because of the unit IDs.

This seems like a problem that could and should easily be fixed at the engine level by applying the concept of "Greater Specificity" from [Cascading Style Sheets](https://www.w3schools.com/css/css_specificity.asp) - a `unit_type` defined in an add-on the player is running should automatically override any core `unit_type` with the same ID.

Would solve other notorious problems too. For example, the backlash over the 1.18 unit balance changes may never have happened if add-on developers only had to copy the old unit configuration files from 1.16 core into their own add-ons when updating to 1.18.

**Comment by knyghtmare** [comment]
> This seems like a problem that could and should easily be fixed at the engine level by applying the concept of "Greater Specificity" from [Cascading Style Sheets](https://www.w3schools.com/css/css_specificity.asp) - a unit_type defined in an add-on the player is running should automatically override any core unit_type with the same ID.

Won't solve the follow-up problem. Players that know the current core stats of those units, playing that addon campaign in question will then try to specify it as a new bug that the damage/attack numbers don't match.

> I realize it's hard to commit unpaid time and effort to something that might not go anywhere, but new core units generally need artists and effort, not internet archeologists and scavenging.

A refined answer, at last.

**Comment by Gothyoba** [comment]
I don’t have much else to add here. My main goal, except for suggesting Dwarvish Witnesses (which don’t seem to have much interest around them) was to bring up some older proposals so that others can decide if a few may be worth pursuing. My (probably) final idea here is to mainline quintains. I now it was rejected before, but the new HttT remake does include them so it might be worth reconsidering.

> I realize it's hard to commit unpaid time and effort to something that might not go anywhere, but new core units generally need artists and effort, not internet archeologists and scavenging.

Agreed.

**Comment by CelticMinstrel** [comment]
> I think a boar can be a male pig or the wild ancestor of domesticated pigs. I might be wrong though. I was personally referring to a wild pig. I suppose they would still be pigs, but they are slightly different from domesticated varieties.

Wild pigs may be slightly different from domesticated pigs, but both are still pigs, so it would feel redundant for both to be added to Wesnoth. We're not creating a zoo here.

**Comment by cooljeanius** [comment]
> We're not creating a zoo here.

That sounds like a good idea for an add-on, though...

**Comment by Gothyoba** [comment]
> List updated for Owl and Great Owl. Not taking Dwarvish Witness into consideration. A hero unit should stay a hero unit. Unless it's used by multiple mainline campaigns (as is case of Elder Mage and Orcish Leader)

Dwarvish Witnesses are now used in both THoT and HttT.
